src/lib.rs

```
@@ -147,16 +152,76 @@ pub fn build(cli: XcCli) -> Result<Produced> {
     rustup::check_needed(&conf).context("checking rustup targets")?;
     cargo::build(&conf).context("running cargo build")?;
 
-    let libs = lipo::assemble_libs(&conf).context("lipo: assembling libraries")?;
-    xcodebuild::assemble(&conf, libs).context("xcodebuild - assemble libraries")?;
+    let libs = {
+        let conf = &conf;
+        let libs_dir = conf.build_dir.join("libs");
+        fs_err::create_dir_all(&libs_dir)?;
+
+        let mut platform_lib_paths = HashMap::new();
+        if conf.cargo_section.iOS {
+            let lib_paths = lib_paths_for_targets(conf, &conf.cargo_section.iOS_targets)?;
+            platform_lib_paths.insert(ApplePlatform::IOS(Environment::Device), lib_paths);
+        }
+        if conf.cargo_section.simulators {
+            let lib_paths = lib_paths_for_targets(conf, &conf.cargo_section.iOS_simulator_targets)?;
+            platform_lib_paths.insert(ApplePlatform::IOS(Environment::Simulator), lib_paths);
+        }
+        if conf.cargo_section.macOS {
+            let lib_paths = lib_paths_for_targets(conf, &conf.cargo_section.macOS_targets)?;
+            platform_lib_paths.insert(ApplePlatform::MacOS, lib_paths);
+        }
+
+        let ending = conf.lib_type.file_ending();
+        let name = &conf.lib_name.replace('-', "_");
+        let output_lib_name = format!("lib{name}.{ending}");
+
+        crate::core::lipo_create_platform_libraries(
+            &platform_lib_paths,
+            &output_lib_name,
+            &libs_dir,
+        )
+    }
+    .context("lipo: assembling libraries")?;
+
+    let bundle_name = conf.module_name()?;
+    let crate_type = match conf.lib_type {
+        conf::LibType::StaticLib => &core::CrateType::Staticlib,
+        conf::LibType::CDyLib => &core::CrateType::Cdylib,
+    };
+    let framework_paths = libs
+        .into_iter()
+        .filter_map(|(platform, lib_path)| {
+            let include_dir = &conf.cargo_section.include_dir;
```

It would be better to instead of filter_map and `.ok()?` which in effect hides errors, do a map without the .ok and then `.collect::<anyhow::Result<Vec<_>>>()?`, which stop the iteration on the first error and propagates the error message.

---

I'm not sure why you marked this as resolved. You disagree with the comment?

---

Oh, sorry for not explicitly reply this comment. I accepted and updated the code from `filter_map` to `map`, so I marked it resolved. Do I misunderstand your idea?

---

Hm, I'm not sure what happened here. I am sure that yesterday I was looking at the code without these changes but today it is changed. I did a pull this morning but nothing to update. Don't know, it's probably my fault. I was too tired to work yesterday. Anyway, my excuses for the confusion. I'll approve and merge